File: app/src/main/java/com/project/scratchstudio/kith_andoid/UI/Comments/list/CommentAdapter.java
package com.project.scratchstudio.kith_andoid.UI.Comments.list;

import android.app.Activity;
import android.icu.text.DateFormat;
import android.icu.text.SimpleDateFormat;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.project.scratchstudio.kith_andoid.R;
import com.project.scratchstudio.kith_andoid.Service.PicassoCircleTransformation;
import com.project.scratchstudio.kith_andoid.network.model.comment.Comment;
import com.squareup.picasso.MemoryPolicy;
import com.squareup.picasso.Picasso;

import java.text.ParseException;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class CommentAdapter extends RecyclerView.Adapter<CommentHolder> {

    public interface OnItemClickListener {
        void onItemClick(Comment item);
    }

    private List<Comment> commentList = new ArrayList<>();
    private Activity activity;

    public CommentAdapter(Activity activity) {
        this.activity = activity;
    }

    public void setCommentList(List<Comment> list) {
        commentList = list;
    }

    @NonNull
    @Override
    public CommentHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View itemView = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.message_dialog_item_layout, viewGroup, false);
        return new CommentHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull CommentHolder holder, int i) {
        Comment comment = commentList.get(i);
        holder.name.setText(comment.getUser().lastName + " " + comment.getUser().firstName);
        holder.position.setText(comment.getUser().position);
        holder.time.setText(comment.getTimestamp().replaceAll("[-0-9]*\\s", "").replaceAll(":[0-9]*$", ""));

        try {
            DateFormat inputFormat;
            if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.N) {
                inputFormat = new SimpleDateFormat("yyyy-MM-dd");
                DateFormat outputFormat = new SimpleDateFormat("dd.MM.yyyy");
                Date newDateFormat;
                newDateFormat = inputFormat.parse(comment.getTimestamp().replaceAll("\\s.*$", ""));
                String outputDateStr = outputFormat.format(newDateFormat);
                holder.date.setText(outputDateStr);
            } else {
                holder.date.setText(comment.getTimestamp().replaceAll("\\s.*$", ""));
            }

        } catch (ParseException e) {
            e.printStackTrace();
        }


        holder.comment.setText(comment.getMessage());
        Picasso.with(activity).load(comment.getUser().photo.replaceAll("@[0-9]*", ""))
                .transform(new PicassoCircleTransformation())
                .error(com.project.scratchstudio.kith_andoid.R.mipmap.person)
                .memoryPolicy(MemoryPolicy.NO_CACHE, MemoryPolicy.NO_STORE)
                .into(holder.photo);
    }

    @Override
    public int getItemCount() {
        return commentList.size();
    }

}

File: app/src/main/java/com/project/scratchstudio/kith_andoid/ui/home_package/Comments/list/CommentAdapter.kt
package com.project.scratchstudio.kith_andoid.ui.home_package.Comments.list

import android.app.Activity
import android.icu.text.DateFormat
import android.icu.text.SimpleDateFormat
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.project.scratchstudio.kith_andoid.R
import com.project.scratchstudio.kith_andoid.service.PicassoCircleTransformation
import com.project.scratchstudio.kith_andoid.network.model.comment.Comment
import com.squareup.picasso.MemoryPolicy
import com.squareup.picasso.Picasso
import java.text.ParseException
import java.util.ArrayList
import java.util.Date

class CommentAdapter(private val activity: Activity) : RecyclerView.Adapter<CommentHolder>() {

    private var commentList: List<Comment> = ArrayList()

    interface OnItemClickListener {
        fun onItemClick(item: Comment)
    }

    fun setCommentList(list: List<Comment>) {
        commentList = list
    }

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): CommentHolder {
        val itemView = LayoutInflater.from(viewGroup.context).inflate(R.layout.message_dialog_item_layout, viewGroup, false)
        return CommentHolder(itemView)
    }

    override fun onBindViewHolder(holder: CommentHolder, i: Int) {
        val comment = commentList[i]
        holder.name.text = comment.user!!.lastName + " " + comment.user!!.firstName
        holder.position.text = comment.user!!.position
        holder.time.text = comment.timestamp!!.replace("[-0-9]*\\s".toRegex(), "").replace(":[0-9]*$".toRegex(), "")

        try {
            val inputFormat: DateFormat
            if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.N) {
                inputFormat = SimpleDateFormat("yyyy-MM-dd")
                val outputFormat = SimpleDateFormat("dd.MM.yyyy")
                val newDateFormat: Date = inputFormat.parse(comment.timestamp!!.replace("\\s.*$".toRegex(), ""))
                val outputDateStr = outputFormat.format(newDateFormat)
                holder.date.text = outputDateStr
            } else {
                holder.date.setText(comment.timestamp!!.replace("\\s.*$".toRegex(), ""))
            }

        } catch (e: ParseException) {
            e.printStackTrace()
        }


        holder.comment.text = comment.message
        Picasso.with(activity).load(comment.user!!.photo!!.replace("@[0-9]*".toRegex(), ""))
                .transform(PicassoCircleTransformation())
                .error(R.mipmap.person)
                .memoryPolicy(MemoryPolicy.NO_CACHE, MemoryPolicy.NO_STORE)
                .into(holder.photo)
    }

    override fun getItemCount(): Int {
        return commentList.size
    }

}
